File: Java/String1/FirstTwo.java
/*
 * CodingBat Java Activity "firstTwo" from String-1 section.
 * codingbat.com
 */

public class FirstTwo 
{
	private static int failures = 0;
	private static final String METHOD_NAME = "firstTwo";
	
	/*
	 * INSTRUCTIONS
	 * Given a string, return the string made of its first two chars, so the 
	 * String "Hello" yields "He". If the string is shorter than length 2, 
	 * return whatever there is, so "X" yields "X", and the empty string "" 
	 * yields the empty string "". Note that str.length() returns the length of
	 * a string.
	 * 
	 * firstTwo("Hello") -> "He"
	 * firstTwo("abcdefg") -> "ab"
	 * firstTwo("ab") -> "ab"
	 */
	public static String firstTwo(String str)
	{
		//put your code ONLY here!
	}
	
	public static void main(String[] args)
	{
		testCase("\"Hello\"", "\"He\"");
		testCase("\"abcdefg\"", "\"ab\"");
		testCase("\"ab\"", "\"ab\"");
		testCase("\"a\"", "\"a\"");
		testCase("\"\"", "\"\"");
		testCase("\"Kitten\"", "\"Ki\"");
		testCase("\"hi\"", "\"hi\"");
		testCase("\"hiya\"", "\"hi\"");
		
		System.out.println();
		System.out.println(failures + " test cases failed.");
	}

	private static void testCase(String a, String result)
	{
		String r;
		try
		{
			r = ""+ "\"" + firstTwo(a.substring(1,a.length()-1)) + "\"";
			if (r.equals(""+result))
			{
				System.out.print("PASS: ");
			}
			else
			{
				System.out.println("FAIL: ");
				failures++;
			}
		}
		catch (Exception e)
		{
			r = e.getMessage();
			System.out.print("FAIL: ");
			failures++;
		}
		System.out.println(METHOD_NAME+"("+a+") -> "+r + ", EXPECTED: "+result);
	}

}

File: Python/string-1/first_two.py
'''
CodingBat Python Activity "first_two" from string-1.
codingbat.com
'''

failures = 0

def first_two(str):
    '''
Given a string, return the string made of its first two chars, so the String "Hello" yields "He".
If the string is shorter than length 2, return whatever there is, so "X" yields "X", and the empty string "" yields the empty string "".

first_two('Hello') → 'He'
first_two('abcdefg') → 'ab'
first_two('ab') → 'ab'
'''
    pass #Replace pass with your code to implement this function





def catchFailures(testCase, result):
    global failures
    
    try:
        assert eval(testCase) == result, "Test Case Failed: "+testCase+ " → "+str(result)
    except Exception as e:
        failures += 1
        print(e)
            


def test():
    global failures
    
    catchFailures("first_two('Hello')", 'He')
    catchFailures("first_two('abcdefg')", 'ab')
    catchFailures("first_two('ab')", 'ab')
    catchFailures("first_two('a')", 'a')
    catchFailures("first_two('')", '')
    catchFailures("first_two('Kitten')", 'Ki')
    catchFailures("first_two('hi')", 'hi')
    catchFailures("first_two('hiya')", 'hi')

        
    if failures > 0:
        print("\r\n "+ str(failures) + " failed test case(s)   :(")
    else:
        print("Good job -- problem solved!")


test()
